lib/shopify_app/sessions_concern.rb

```diff
@@ -87,5 +88,16 @@ def return_address
       session.delete(:return_to) || main_app.root_url
     end
 
+    def perform_after_authenticate_job
+      config = ShopifyApp.configuration.after_authenticate_job
+
+      return unless config && config[:job].present? && config[:inline].present?
+
+      if config[:inline]
```

Comment:
Have a look at the [`SessionControllerTests`](https://github.com/Shopify/shopify_app/blob/master/test/controllers/sessions_controller_test.rb#L97-L106). One way to do this is by testing the `#callback` method.